devito/types/dimension.py

```diff
@@ -607,6 +607,65 @@ def middle(cls, name, parent, thickness_left, thickness_right, local=False):
                    thickness=((lst, thickness_left), (rst, thickness_right)),
                    local=local)
 
+    @classmethod
+    def intersection(cls, sdims):
+        # Check all SubDimensions supplied overlap
+        # NOTE: Think the test for overlap isn't strict enough?
+        if not all([sdims[0].overlap(sdim) for sdim in sdims[1:]]):
+            raise ValueError("SubDimensions do not all overlap")
+
+        name = '_n_'.join([sdim.name for sdim in sdims])
+        try:
+            [parent] = set([sdim.parent for sdim in sdims])
```

Comment:
{self.parent}.union(*[sdim.parent for sdim in sdims])

Reply:
Note that this is both getting the parent and checking they all share a parent, since the unpack will fail with `ValueError` if the set does not contain one item